src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -16,6 +23,35 @@ public void setup() {
 
     @Test
     public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+        Product product = productDao.save(new Product("test", "test", new BigDecimal(100), usd, 0, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg"));
```

Comment:
save the collection in setUp() to dao and just get its Product instances
no need to call save everywhere